### ISSUE TITLE
[Feature request] Wiimotes as guns in RPCS3

### ISSUE THREAD

**Issue by EmuAl** [RPCS3/rpcs3#15165]
### Quick summary

The ability to use wiimotes as guns for rail shooters in rpcs3.
Currently in a roundabout way you can use a wiimote with mayflash bar in mouse mode (mode 2) for single player in games like time crisis, razing storm etc. With other emulators like Flycast, Mame and Teknoparrot you can use the wiimote with mayflash bar in Joystick mode (mode 3) to give up to 4 player ability in games in these emulators.

### Details

**What part of RPCS3 would be affected by your feature?**
This addition would affect gameplay of playstation move gun games

**Why your feature is important to RPCS3.**
It would allow the playstation move gun games to be played like they were meant to be played.

**Details**
I will try to explain this but I feel like this is really close to working already and just needs some work in the right places or I could be wrong and its not possible with rpcs3 at all, so here goes.

The game i'm using to test and describe the details and workings here is Time Crisis: Razing Storm.

In order to get any input in the game with the wiimote & mayflash bar in joystick mode (mode 3) I have to set the input to Handlers: MMJoystick, when i try Handlers: SDL, I can set up the wiimote but I get no input in the game.

![Pads Panel](https://github.com/RPCS3/rpcs3/assets/47227078/55f95261-6e3a-4e12-9a0f-4f506659f8e3)

![USB Device - PS Move (Fake) Panel](https://github.com/RPCS3/rpcs3/assets/47227078/a705f0df-f111-4a07-b695-a5fbe65717ec)

![config](https://github.com/RPCS3/rpcs3/assets/47227078/eff3ebed-ce3b-49fb-acd6-c97a0b620bba)

In the first image you can see I have set the controls as required by the PS Move controls in the second image and in the third image the config has the correct I/O settings to enable the PS Move controller
 
https://streamable.com/4w1jw6

In the video above, you can see I have the correct style of movement using the wiimote it moves around like its a gun aiming at the screen

https://streamable.com/qd7xya

In the second video you can see I have the same movement inside the game with the move controller crosshair, in the second part of this video you can see what it is like in game, the game gun crosshair doesn't follow the move crosshair, its like the move crosshair has to pass the centre of the screen in order for the gun to change direction, then it travels in a straight line from there. This is relative movement as opposed to absolute movement, which i think is the problem. The gun input for joystick movement must be absolute for it to follow the move crosshair. In the other emulators I mention above they use the absolute style of movement and I can setup the wiimote as a gun with them and it works really well. 

In the following video you can see it working in the other emulators.

https://streamable.com/vsyril

to recap, my feature request is to have absolute input as a selectable option on the controller setup panel to allow for the correct input in RPCS3 gun games

If you need any more info please ask.

Thanks for your time and hopefully this is possible.

**Comment by SeongGino** [comment]
FWIW it isn't just for Wiimote users where this would be useful; [other lightgun solutions](https://github.com/SeongGino/ir-light-gun-plus) also offers the option to output the gun's current position as an analog stick/button as gamepad press events.

Hopefully someone interested will pick up on this in the future.

**Comment by Megamouse** [comment]
I don't see how this is related to the wiimote. You clearly have proper absolute input.
I don't know where the weird input lag is coming from. Could be a performance issue.
You didn't upload a log anyway.
Closing as implemented.
